### PR TITLE
Disable OrganizeImports commands if they are not available

### PR DESCRIPTION
**Pull request by wjoel**
Rather than having Metals show a warning message, leave it to the client to decide on what to do with the reason for why they are disabled, and if/how it should be displayed.

Fixes https://github.com/scalameta/metals-zed/issues/14

Then it looks like this in VS Code:
organize-imports-disabled-vscode

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala [hunk 1]
```diff
@@ -131,21 +127,23 @@ class OrganizeImportsQuickFix(
...
     val hasUnused = params
       .getContext()
       .getDiagnostics()
       .asScala
       .collect { case ScalacDiagnostic.UnusedImport(name) => name }
       .nonEmpty
 
-    if (hasUnused && !diagnostics.hasDiagnosticError(file)) {
-      true
+    if (!hasUnused) {
```
Comment (wjoel) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala:141]: This should arguably also check `if (!isScalaOrSbt(file))`, as the other one does?

Comment (tgodzik) [reply metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala:141]: That would make sense, though I don't think Metals will be invoked for anything else than Scala and Java files. Maybe we should be explicit about not doing organize import on Java

File: metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala [hunk 2]
```diff
@@ -85,33 +96,18 @@ class SourceOrganizeImports(
...
-      } else {
-        true
-      }
+  ): Option[String] =
+    if (!isScalaOrSbt(file)) {
+      Some("Only supported for Scala and sbt files")
+    } else if (diagnostics.hasDiagnosticError(file)) {
+      scribe.info(canNotOrganizeImportsWithErrors)
+      Some(canNotOrganizeImportsWithErrors)
     } else {
```
Comment (tgodzik) [inline metals/src/main/scala/scala/meta/internal/metals/codeactions/OrganizeImports.scala:108]: We are now missing one of the conditions `isSourceOrganizeImportCalled(params)` and organize imports action is returned everywhere. We want to return it on unused imports and also when explicitly called.  Maybe we should still keep isCallAllowed for that and also check that flag?

**Review by tgodzik — CHANGES_REQUESTED** [review]
Thanks for looking into this! I wasn't aware we had the disabled flag before.

**Comment by wjoel** [comment]
It'll be a week or two before I'll have time to get back to this, so if anyone else wants to address the comments, please feel free to do so.

Otherwise I will get it done (and Zed is being a bit annoying even with those changes, but otherwise it's quite fun overall, so I have some strong personal motivations), but definitely not this week.

**Comment by tgodzik** [comment]
> It'll be a week or two before I'll have time to get back to this, so if anyone else wants to address the comments, please feel free to do so.
> 
> Otherwise I will get it done (and Zed is being a bit annoying even with those changes, but otherwise it's quite fun overall, so I have some strong personal motivations), but definitely not this week.

There is no hurry! Just let us know if you have any problems or are unable to finish it.

**Review by tgodzik — APPROVED** [review]
Thanks! I pushed a minor fix to the tests to include information if an action is disabled

**Pull request merged**